### ISSUE TITLE
Character names issue

### ISSUE THREAD

**Issue by Regner** [Kyria/EsiPy#7]
Not sure if this is an issue with ESI, PySwagger, or EsiPy.

I am simply attempting to convert some character IDs to names. Here is some code to reproduce the problem I am having:

```python
from esipy import App, EsiClient

app = App.create('https://esi.tech.ccp.is/latest/swagger.json')
client = EsiClient()

client.request(app.op['get_characters_names'](character_ids=[90000001]))
```

```
Traceback (most recent call last):
  File "/Users/regnerblokandersen/Library/Application Support/IntelliJIdea2016.3/python/helpers/pydev/pydevd.py", line 1588, in <module>
    globals = debugger.run(setup['file'], None, None, is_module)
  File "/Users/regnerblokandersen/Library/Application Support/IntelliJIdea2016.3/python/helpers/pydev/pydevd.py", line 971, in run
    pydev_imports.execfile(file, globals, locals)  # execute the script
  File "/Users/regnerblokandersen/Library/Application Support/IntelliJIdea2016.3/python/helpers/pydev/_pydev_imps/_pydev_execfile.py", line 18, in execfile
    exec(compile(contents+"\n", file, 'exec'), glob, loc)
  File "/Users/regnerblokandersen/git/personal/capsuleers/repro_case.py", line 6, in <module>
    client.request(app.op['get_characters_names'](character_ids=[90000001]))
  File "/Users/regnerblokandersen/virtual_envs/capsuleers/lib/python3.6/site-packages/pyswagger/spec/v2_0/objects.py", line 283, in __call__
    _convert_parameter(final(p))
  File "/Users/regnerblokandersen/virtual_envs/capsuleers/lib/python3.6/site-packages/pyswagger/spec/v2_0/objects.py", line 274, in _convert_parameter
    params[i].extend([tuple([p.name, v]) for v in c.to_url()])
  File "/Users/regnerblokandersen/virtual_envs/capsuleers/lib/python3.6/site-packages/pyswagger/primitives/_array.py", line 79, in to_url
    return [str(self)]
  File "/Users/regnerblokandersen/virtual_envs/capsuleers/lib/python3.6/site-packages/pyswagger/primitives/_array.py", line 70, in __str__
    raise SchemaError('Unsupported collection format when converting to str: {0}'.format(self.__collection_format))
pyswagger.errs.SchemaError: Unsupported collection format when converting to str: None
```

Would love any help you can offer.

**Comment by Kyria** [comment]
So, here's what i've found so far (by digging into pyswagger lol)

Looks like it's a pyswagger issue (i'm going to open a ticket there), as by default in swagger, arrays have a "collectionFormat" set to csv, but here, for an unknown reason it's always to "None"

Currently CCP (in ESI) do not set that value (as the default 'csv' is what they need), but something in pyswagger bugs, so when it tries to get the format (with a `getattr(obj, 'collectionFormat', 'csv')`), it always return "None" instead of 'csv'. That's why we have that issue.

So far, I've found a way to bypass this issue, **but it may break other things** (as I didn't test it with anything else). You'll just have to add these two line before creating the App:
```python
from pyswagger.spec.v2_0.objects import Parameter
Parameter.__swagger_fields__['collectionFormat']='csv'
```

And the result for your endpoint :
```python
>>> from pyswagger.spec.v2_0.objects import Parameter
>>> Parameter.__swagger_fields__['collectionFormat']='csv'

>>> from esipy import App, EsiClient

>>> app = App.create('https://esi.tech.ccp.is/latest/swagger.json')
>>> client = EsiClient()

>>> res=client.request(app.op['get_characters_names'](character_ids=[90000001]))
>>> res.data
[{u'character_id': 90000001, u'character_name': u'Vert1'}]
```

I'll keep you informed if I have other infos about this.

**Comment by Regner** [comment]
Yea, that was basically the same conclusion I came to as well and the same solution. Will continue with that for now. Will follow the pyswagger issue.

**Comment by Kyria** [comment]
@Regner looks like pyswagger just released a 0.8.25 version with a fix for this issue.
It looks like it fixed the problem:
```python
>>> from esipy import App, EsiClient
>>> app = App.create('https://esi.tech.ccp.is/latest/swagger.json')
>>> client = EsiClient()
>>> res=client.request(app.op['get_characters_names'](character_ids=[90000001]))
>>> res.data
[{u'character_id': 90000001, u'character_name': u'Vert1'}]
```

**Comment by Regner** [comment]
Thanks for looking into that. :)
